recipes/pistache/all/patches/0001-promote-cpp17.patch

```diff
@@ -0,0 +1,282 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index ce10678..a623102 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -3,7 +3,7 @@ cmake_minimum_required (VERSION 3.4)
+ set(CMAKE_BUILD_TYPE_INIT Release)
+ 
+ # CMAKE Pin cxx compiler to CXX14 until update to CXX17
+-set(CMAKE_CXX_STANDARD 14)
++set(CMAKE_CXX_STANDARD 17)
+ 
+ set(CMAKE_CXX_STANDARD_REQUIRED ON)
+ set(CMAKE_CXX_EXTENSIONS OFF)
+diff --git a/include/pistache/router.h b/include/pistache/router.h
+index 1c8765e..f3834eb 100644
+--- a/include/pistache/router.h
++++ b/include/pistache/router.h
+@@ -12,12 +12,12 @@
+ #include <tuple>
+ #include <unordered_map>
+ #include <vector>
++#include <string_view>
+ 
+ #include <pistache/flags.h>
+ #include <pistache/http.h>
+ #include <pistache/http_defs.h>
+ 
+-#include "pistache/string_view.h"
```

Comment:
seems like the patch reduces the library's original compatibility and number of supported configurations.
e.g. it contained its own `std::string_view` implementation for pre-C++17 compilers, but now it will always require C++17

Reply:
This change is kind of required due https://github.com/pistacheio/pistache/issues/835

Pistache will update to C++17 as requirement. If we use its version, clang won't work: https://github.com/conan-io/conan-center-index/pull/3719#issuecomment-734956306

Reply:
I'd just raise `ConanInvalidConfiguration` for failing clang version(s) and keep the amount of patched source code as minimal as possible.

Reply:
@SSE4 Done.